flashpay/apps/payments/views.py

```diff
@@ -131,3 +138,117 @@ def list(self, request: Request, *args: Any, **kwargs: Any) -> Response:
             },
             status.HTTP_200_OK,
         )
+
+
+class VerifyTransactionView(GenericAPIView):
+    permission_classes = [IsAuthenticated]
+    serializer_class = VerifyTransactionSerializer
+
+    def post(self, request: Request, format: Any = None) -> Response:
+        indexer_client = settings.INDEXER_CLIENT
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        txid = serializer.validated_data["txid"]
+
+        try:
+            api_response = indexer_client.transaction(
+                txid=txid,
+            )
+        except IndexerHTTPError as e:
+            logger.error(
+                f"Error finding transaction with"
+                f'transaction id: {serializer.validated_data["txid"]}'
+                f"due to: {str(e)}"
+            )
+            return Response(
+                data={
+                    "status_code": status.HTTP_500_INTERNAL_SERVER_ERROR,
+                    "message": "Could not find transaction",
+                    "data": None,
+                },
+                status=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            )
+
+        tx_note = api_response["transaction"].get("note")
+        txn_ref = b64decode(tx_note).decode()
+        transactions = Transaction.objects.filter(
+            txn_ref=txn_ref, status=TransactionStatus.PENDING
+        )
+
+        if not transactions.exists():
+            return Response(
+                data={
+                    "status_code": status.HTTP_400_BAD_REQUEST,
+                    "message": "Transaction does not exist",  # noqa: E501
+                    "data": None,
+                },
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        transaction = transactions.first()
+
+        # check if transaction type of 'axfer' i.e transferring algo assets.
+        if api_response["transaction"]["tx-type"] != "axfer":
+            return Response(
+                data={
+                    "status_code": status.HTTP_400_BAD_REQUEST,
+                    "message": "An error occured while setting up your account due to bad transaction provided.",  # noqa: E501
+                    "data": None,
+                },
+                status=status.HTTP_400_BAD_REQUEST,
+            )
```

Comment:
What does this block mean?
Are you trying to prevent sending `Non-Algorand assets`? Is that a possibility?

Reply:
I only checked for Algorand assets transfer, I should be checking for both `pay` and `axfer' transactions.